src/bitbutler.sh

```diff
@@ -698,6 +701,69 @@ JSON
   esac
 }
 
+function selfupdate() {
+  REPOSITORY_SLUG="yiisoft/yii2"
+  SELFUPDATE_WORKING_DIR='/tmp/bitbutler-selfupdate'
+  SETTINGS_FILE="$BB_VENDOR_PATH/make.settings"
+
+  latest_tag_json=$(curl \
+    --silent \
+    --header "Accept: application/vnd.github.v3+json" \
+    "https://api.github.com/repos/${REPOSITORY_SLUG}/releases" |
+    jq '.[0]')
+
+  if [[ "$latest_tag_json" == "null" ]]; then
+    die "No releases yet, cannot update"
+  fi
+
+  latest_tag_name=$(jq --raw-output '.tag_name' <<<"$latest_tag_json")
+
+  v "Current script version: ${SCRIPT_VERSION}"
+  v "Latest version is: $latest_tag_name"
+  if [[ "$latest_tag_name" != "${SCRIPT_VERSION}" ]]; then
+    l "Your version is outdated."
+  else
+    l "You already have the lasted version."
+    exit 0
+  fi
+
+  confirm "Do you want to update now?"
+
+  tarball_url=$(jq --raw-output '.tarball_url' <<<"${latest_tag_json}")
+  tarball_path="${SELFUPDATE_WORKING_DIR}/tarball.tar.gz"
+
+  rm -r "${SELFUPDATE_WORKING_DIR}"
```

Comment:
I think this `rm` should have the `-f` flag as well. It avoids potential errors or inteactivity